SCClassLibrary/Common/Core/Object.sc

```diff
@@ -217,6 +217,17 @@ Object  {
 	hash { _ObjectHash; ^this.primitiveFailed }
 	identityHash { _ObjectHash; ^this.primitiveFailed }
 
+	// lazy equality: same as == for objects
+	// "composed" for lazy operands (patterns, UGens)
+	|==| { |that|
```

Comment:
Is there a reason why you can't reuse `reverseComposeBinaryOp`?

Reply:
Yes, there is a reason. You can't guarantee that one operand of `|==|` is an abstract function.

Take `1 |==| 2`. First it enters `Object:|==|`. This hasn't taken into account the class of the right-hand operand. If it directly calls `2.reverseComposeBinaryOp('|==|', 1)`, it fails because `reverseComposeBinaryOp` is not defined for simple numbers. So we need a compose method specifically for this case.

(We have to compose equality checks differently from other binary operators. In `1 + { 2.rand }`, it calls `reverseComposeBinaryOp` because `Integer:+` failed first. But an equality check doesn't fail -- `1 == { 2.rand }` is just false.)

Reply:
Ah yes, I understand.